kernel/src/thread/mod.rs

```diff
@@ -45,20 +45,18 @@ impl Thread {
     /// This function returns `None` if the current task is not associated with
     /// a thread, or if called within the bootstrap context.
     pub fn current() -> Option<Arc<Self>> {
-        Task::current()?
-            .data()
-            .downcast_ref::<Weak<Thread>>()?
-            .upgrade()
+        Self::from_task_data(&Task::current()?).cloned()
     }
 
-    pub(in crate::thread) fn task(&self) -> &Arc<Task> {
-        &self.task
+    /// Gets the Thread from task's data
+    pub fn from_task_data(task: &Arc<Task>) -> Option<&Arc<Self>> {
```

Comment:
First of all, I suggest calling it `borrow_from_task` to make it more clear that the returned value is a reference, not an owned object.

Second, I suggest changing the return type from `Option<&Arc<Self>>` to `&Arc<Self>`. I checked all user code, they either call `unwrap` explicitly or assume that the conversion succeeds implicitly.